Add decky-zerotier v0.3.0

hyacz opened the pull request:
<!-- Make sure to include your plugin name below! -->

# Decky Zerotier

https://github.com/hyacz/decky-zerotier
https://github.com/SteamDeckHomebrew/decky-plugin-database/pull/665

This plugin is a [Zerotier](https://www.zerotier.com/) client UI designed for SteamDeck to support the LAN connection function of some games.

Compared to v0.1.0, the Zerotier core binary has been upgraded from v1.14.0 to v1.14.1, and it is now built using Github action.

## Checklist:

### Developer Checklist

- [x] I am the original author or an authorized maintainer of this plugin.
- [x] I have abided by the licenses of the libraries I am utilizing, including attaching license notices where appropriate.

### Plugin Checklist

- [x] I have verified that my plugin works properly on the Stable and Beta update channels of SteamOS.
- [x] I have verified my plugin is unique or alternatively provides more/alternative functionality to a similar plugin already on the store.

<!-- The following section needs to be modified as yes/no answers by the plugin developer. -->

<!-- Ex: "**Yes/No**: ..." becomes "**Yes**: ..." -->

### Plugin Backend Checklist

- **Yes**: I am using a custom backend other than Python.
- **Yes**: I am using a tool or software from a 3rd party FOSS project that does not have it's dependencies [statically linked](https://en.wikipedia.org/wiki/Static_library).
- **No**: I am using a custom binary that has all of it's dependencies statically linked.

<!-- The following section is should be modified to fit the conditions for plugin testing found here: https://wiki.deckbrew.xyz/en/plugin-dev/review-and-testing -->

## Testing

<!-- Remove this box for SteamOS Preview testing if you do not use a custom or remote binary, for more info follow the URL in the comment above the testing section. -->
- [ ] Tested on SteamOS Preview Update Channel.

TrainDoctor reviewed (approved):
Code review was simple as it's just a few more additions to the original submission. In future please include a link to the comparison between your original PR and this one (commit wise) so I can review the changes between then and now.
Good for testing again.

TrainDoctor commented on .gitmodules:
```diff
@@ -219,3 +219,6 @@
 [submodule "plugins/DeckyInhibitScreenSaver"]
 	path = plugins/DeckyInhibitScreenSaver
 	url = https://github.com/xfangfang/DeckyInhibitScreenSaver
+[submodule "plugins/plugins/decky-zerotier"]
+	path = plugins/plugins/decky-zerotier
+	url = https://github.com/hyacz/decky-zerotier
```
Just realized that this needs to be fixed.
```suggestion
[submodule "plugins/decky-zerotier"]
	path = plugins/decky-zerotier
	url = https://github.com/hyacz/decky-zerotier
```

TrainDoctor reviewed (dismissed):
Completely missed this, this won't be caught and built by CI until the git submodule is fixed.

hyacz commented:
Sorry, I made a mistake, I am not familiar with PR process. I used a force push to make the Commit Message easier to read, I hope this will not cause more problems.

hyacz commented:
I re-corrected the submodule so that it now correctly points to decky zerotier tag v0.2.0

hyacz commented:
https://github.com/hyacz/decky-zerotier/compare/v0.2.0...v0.2.1

- Updated dependency packages
- Fixed remote_binary issue (thanks to eXhumer https://discord.com/channels/960281551428522045/960284327445418044/threads/1271103396484485273)
- Fixed SHA256 of remote_binary, caused rebuild of zerotier-one due to incorrectly triggered CI. (https://github.com/hyacz/decky-zerotier/actions/runs/11528581986)

hyacz commented:
https://github.com/hyacz/decky-zerotier/compare/v0.2.1...v0.3.0

- Updated Zerotier from v1.14.1 to v1.14.2
- Updated dependency packages

TrainDoctor commented:
Closing as untested. If you are able to locate a tester in future we will happily accept a re-submission.

Deadboy666 commented:
i might be stupid but i cant remember the chmod command to get zeroteir to work it was working about a month ago but tried updating the plugin and it reset the permissions on it
great plugin btw

hyacz commented:
Sorry, I was a little busy some time ago and didn't have time to deal with this job. How can I get someone to help me test the plug-in?

TrainDoctor commented:
> Sorry, I was a little busy some time ago and didn't have time to deal with this job. How can I get someone to help me test the plug-in?

@hyacz, we are now having plugin developers cooperate to help test each others plugins so that could help you.
For now, we have a potential tester here.
@Deadboy666, are you willing to submit a [testing report](decky.xyz/testing#submitting-a-test-report) for this plugin?

Deadboy666 commented:
![Screenshot_20250208_171853](https://github.com/user-attachments/assets/03165eaa-720e-4023-a790-16762578b09d)
version 0.3 works

TrainDoctor commented:
> it works now i just needed to revert to decky stable and stable steamos Requires manualy setting up premissions for the zerotier binary but then it works

@Deadboy666 Please follow the testing report guide otherwise we can't accept it, but please wait to complete that report once this PR has been updated to fix the plugin so the install is working out of the box.
@hyacz, please make any needed fixes and then update this PR with those changes.

hyacz commented:
New PR #761  has been submitted

https://github.com/hyacz/decky-zerotier/compare/v0.3.0...v0.3.1
- Dependency updates managed by DependBot
- Binary file permissions have been fixed via https://github.com/hyacz/decky-zerotier/commit/741fda73d94a70e0562daeb086a731c0a1233b6b
